rclpy/src/rclpy_common/src/common.c

```diff
@@ -109,8 +117,48 @@ rclpy_convert_to_py_names_and_types(rcl_names_and_types_t * names_and_types)
   return pynames_and_types;
 }
 
+static
 PyObject *
-rclpy_convert_to_py_qos_policy(void * profile)
+_convert_rmw_time_to_py_duration(const rmw_time_t * duration)
+{
+  uint64_t total_nanoseconds = RCUTILS_S_TO_NS(duration->sec) + duration->nsec;
+  PyObject * pyduration_module = NULL;
+  PyObject * pyduration_class = NULL;
+  PyObject * args = NULL;
+  PyObject * kwargs = NULL;
+  PyObject * pyduration_object = NULL;
+
+  pyduration_module = PyImport_ImportModule("rclpy.duration");
+  if (!pyduration_module) {
+    goto cleanup;
+  }
+  pyduration_class = PyObject_GetAttrString(pyduration_module, "Duration");
+  if (!pyduration_class) {
+    goto cleanup;
+  }
+  args = PyTuple_New(0);
+  if (!args) {
+    goto cleanup;
+  }
+  kwargs = Py_BuildValue("{s:l}", "nanoseconds", total_nanoseconds);
```

Comment:
@emersonknapp changing [lowercase `l` to uppercase `K`](https://docs.python.org/3/c-api/arg.html#c.Py_BuildValue) may fix the windows issue. Using gcc 7.3 `long` and `long long` are both 8 bytes, but on windows `long` is only 4 bytes https://docs.microsoft.com/en-us/cpp/cpp/data-type-ranges?view=vs-2019

Reply:
Also, the documentation for `Py_BuildValue` doesn't include a `:` for dictionaries, it just says consecutive C values become key/value pairs. I think this should read `{sK}`.

Reply:
Thanks - I forgot about the `K` specifier, C99 spec only guarantees 32 bit for `long`, so that makes perfect sense, I did want a `long long`